packages/fast-components-react-msft/src/pivot/pivot.tsx

```diff
@@ -0,0 +1,240 @@
+import * as React from "react";
+import { get } from "lodash-es";
+import * as ReactDOM from "react-dom";
+import { canUseDOM } from "exenv-es6";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    PivotHandledProps,
+    PivotManagedClasses,
+    PivotUnhandledProps,
+} from "./pivot.props";
+import {
+    ManagedClasses,
+    PivotClassNameContract,
+} from "@microsoft/fast-components-class-name-contracts-msft";
+import { Direction } from "@microsoft/fast-application-utilities";
+import { toPx } from "@microsoft/fast-jss-utilities";
+import { TabsClassNameContract } from "@microsoft/fast-components-react-base";
+import { Tabs as BaseTabs } from "@microsoft/fast-components-react-base";
+import { PivotProps } from ".";
+
+export interface PivotState {
+    offsetX: number;
+    activeId: string;
+    tabPanelIndex: number;
+}
+
+class Pivot extends Foundation<PivotHandledProps, PivotUnhandledProps, PivotState> {
+    public static displayName: string = "Pivot";
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(
+        nextProps: PivotProps,
+        prevState: PivotState
+    ): null | Partial<PivotState> {
+        if (nextProps.activeId && nextProps.activeId !== prevState.activeId) {
+            return {
+                activeId: nextProps.activeId,
+            };
+        }
+
+        return null;
+    }
+
+    protected handledProps: HandledProps<PivotHandledProps> = {
+        label: void 0,
+        items: void 0,
+        managedClasses: void 0,
+    };
+
+    private tabsRef: React.RefObject<any>;
+
+    private ltr: Direction;
+
+    private prevTabPanelIndex: number;
+
+    /**
+     * The constructor
+     */
+    constructor(props: PivotProps) {
+        super(props);
+
+        if (Array.isArray(this.props.items)) {
+            this.state = {
+                offsetX: 0,
+                tabPanelIndex: 0,
+                activeId:
+                    typeof this.props.activeId === "string"
+                        ? this.props.activeId
+                        : get(this.props.items[0], "id", ""),
+            };
+        }
+
+        this.tabsRef = React.createRef();
+    }
+
+    public componentDidMount(): void {
+        this.ltr = this.getLTR();
+        this.setActiveIndicatorOffset();
+        this.prevTabPanelIndex = this.state.tabPanelIndex;
+    }
+
+    public componentDidUpdate(prevProps: PivotProps, prevState: PivotState): void {
+        if (this.ltr !== this.getLTR()) {
+            this.setActiveIndicatorOffset();
+            this.ltr = this.getLTR();
+        }
+
+        if (this.state.activeId !== prevState.activeId) {
+            this.setActiveIndicatorOffset();
+            this.updateTabPanelIndex();
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLButtonElement | HTMLAnchorElement> {

```

Comment:
I don't think the function signature is accurate here - does this actually render a `<button />` or `<a />`?